spec/index.bs

```diff
@@ -0,0 +1,404 @@
+
+    Issue: Nothing is limiting fetch requests to be HTTPS only.
+  </section>
+</section>
+
+<section>
+  <h2 id="api-description">API Description</h2>
+
+  <section>
+    <h3 id="service-worker-registration-extensions">Extensions to the {{ServiceWorkerRegistration}} interface</h3>
+
+    <pre class="idl">
+      partial interface ServiceWorkerRegistration {
+        readonly attribute SyncManager sync;
+      };
+    </pre>
+
+    The <code><dfn attribute for=SyncManager title=sync>sync</dfn></code> attribute exposes a {{SyncManager}}, which has an associated <a>service worker registration</a> represented by the {{ServiceWorkerRegistration}} on which the attribute is exposed.
+  </section>
+
+  <section>
+    <h3 id="sync-manager-interface">{{SyncManager}} interface</h3>
+
+    <pre class="idl">
+      [Exposed=(Window,Worker)]
+      interface SyncManager {
+        Promise&lt;SyncRegistration&gt; register(optional SyncRegistrationOptions options);
+        Promise&lt;SyncRegistration&gt; getRegistration(DOMString tag);
+        Promise&lt;sequence&lt;SyncRegistration&gt;&gt; getRegistrations();
+      };
+
+      dictionary SyncRegistrationOptions {
+        DOMString tag;
+      };
+    </pre>
+
+    The <code><dfn method for=SyncManager title="register(options)">register(<var>options</var>)</dfn></code> method, when invoked, MUST return <a>a new promise</a> <var>promise</var> and run the following steps <a>in parallel</a>:
+    <ol>
+      <li>
+        Let <var>serviceWorkerRegistration</var> be the {{SyncManager}}'s associated <a>service worker registration</a>.
+      </li>
+      <li>
+        If the <a>global object</a> specified by the <a>incumbent settings object</a> is a {{WorkerGlobalScope}} instance, and the <var>serviceWorkerRegistration</var>'s <a>active worker</a> is not currently <a lt="control">controlling</a> any <a>clients</a>, <a>reject</a> <var>promise</var> with an {{InvalidAccessError}} and abort these steps.
+      </li>
+      <li>
+        Let <var>currentRegistration</var> be the <a lt="sync registration">registration</a> in <var>serviceWorkerRegistration</var>'s <a>list of sync registrations</a> whose <a>tag</a> equals <var>options.tag</var> if it exists, else null.
+      </li>
+      <li>
+        If <var>currentRegistration</var> is not null, <a>resolve</a> <var>promise</var> with a new {{SyncRegistration>> instance associated with <var>currentRegistration</var> and abort these steps.
```

Comment:
I see that you removed the case where the tags are the same but the options differ. I suppose there are no other options than tag at the moment so it's okay. We'll need to make sure we add back that case if we add further options in the future.